src/Microsoft.AspNetCore.Diagnostics.HealthChecks/HealthCheckAppBuilderExtensions.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Diagnostics.HealthChecks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class HealthCheckAppBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a middleware that provides a REST API for requesting health check status.
+        /// </summary>
+        /// <param name="applicationBuilder">The <see cref="IApplicationBuilder"/></param>
+        /// <param name="path">The path on which to provide the API</param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseHealthChecks(this IApplicationBuilder applicationBuilder, PathString path)
```

Comment:
I don't think the path belongs in ConfigureServices. The middleware and the service are not as tightly coupled as some of our other middleware-service pairs. The service is not tied to HTTP at all, it could be used to drive background monitoring tasks instead of pull-based HTTP endpoints.